### ISSUE TITLE
[no-unused-vars] False positive when destructuring

### ISSUE THREAD

**Issue by sveyret** [typescript-eslint/typescript-eslint#1187]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-vars": [
      "error",
      { "argsIgnorePattern": "^_" }
    ]
  }
}
```

```TS
const value = {
  a: 0,
  b: 'hello',
  c: 1
};
const { a, ...rest } = value;
call_a_function(rest);
```

**Expected Result**

No error, `a` is used only to remove this property from `rest`.

**Actual Result**

'a' is assigned a value but never used. eslint(@typescript-eslint/no-unused-vars)

**Additional Info**

This was working in TSLint.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.6.1` |
| `@typescript-eslint/parser`        | `2.6.1` |
| `TypeScript`                       | `3.6.4` |
| `ESLint`                           | `6.6.0` |
| `node`                             | `12.13.0` |
| `npm`                              | `6.12.0` |

**Comment by bradzacher** [comment]
Note that our rules are not 1:1 replacement with tslint.
Some of our rules are built off of eslint base rules which have been around long before tslint.

You'll find an option to disable this in the rule docs
https://eslint.org/docs/rules/no-unused-vars
